torch/_subclasses/fake_tensor.py

```diff
@@ -1413,6 +1435,19 @@ def _validate_cache_key(
         # Avoid caching for any ops that would require a more sophisticated
         # caching implementation, e.g., data dependent ops or ops that modify
         # the inputs.
+
+        if isinstance(func, torch._ops.HigherOrderOperator):
+            # For invoke_subgraph op, if the identifier is set then its safe to
+            # cache the fake tensor result.
+            from torch._higher_order_ops.utils import (
+                registered_hop_fake_tensor_cache_key_validation_fns,
+            )
+
+            if func in registered_hop_fake_tensor_cache_key_validation_fns:
+                return registered_hop_fake_tensor_cache_key_validation_fns[func](
+                    self, *args, **kwargs
+                )
```

Comment:
Sorry I kind of forgot the discussion we had on Thursday. I remember we reasoned out that everything was safe? If that's the case, can we include the reasoning here?

Reply:
I see, we're going the conservative approach